Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/ApplePayTokenizationViewModel.swift

```diff
@@ -492,7 +518,8 @@ extension ApplePayTokenizationViewModel: PKPaymentAuthorizationControllerDelegat
                                                             from: payment.token.paymentData)
             }
 
-            let billingAddress = clientSessionBillingAddressFromApplePayBillingContact(payment.billingContact)
+            let billingAddress = clientSessionAddressFromApplePayContact(payment.billingContact)
+            let shippingAddress = clientSessionAddressFromApplePayContact(payment.shippingContact)
```

Comment:
For now, I'm just adding the shipping address in the client session.

It might be interesting to retrieve the email, first name, last name and phone number and update them in the client session. I don't know if it's something that's currently supported to update these customer fields?

Reply:
Here is a second PR: https://github.com/primer-io/primer-sdk-ios/pull/947 to update ClientSession with `emailAddress` and `mobileNumber` from `shippingContact`.

Those two fields are oftenly required by ecommerce app to build a fast checkout, so they can handle those informations about customer.